docs/details/main-components/draw.rst

```diff
@@ -64,12 +68,191 @@ Hierarchy of modules
+(using, for example, ``lv_draw_task_get_label_dsc()``) and modified (e.g., change the color, text, font, etc.).
+It also means that in ``LV_EVENT_DRAW_TASK_ADDED`` the ``draw_task``s and draw descriptors are already
+initialized. In practice, it looks like this:
+
+.. code-block:: c
+   /*In LV_EVENT_DRAW_TASK_ADDED*/
+   lv_draw_task_t * t = lv_event_get_draw_task(e);
+   lv_draw_label_dsc_t * draw_dsc =  lv_draw_task_get_label_dsc(t);
+   /*Make the color lighter for longer texts*/
+   draw_dsc->color = lv_color_lighten(draw_dsc->color, LV_MIN(lv_strlen(draw_dsc->text) * 5, 255));
+
+   /*Create new draw tasks if needed here by calling lv_draw_...() functions*/
+
+In the following, a detailed description of each draw task and descriptor type will be shown.
+
+Relation to styles
+------------------
+
+In most of the cases the style properties are 1-1 mapped to the draw descriptor fields.
+For example ``label_dsc.color`` is the ``text_color`` style property.
+Or ``shadow_dsc.width``, ``line_dsc.opa``, ``arc_dsc.width`` are
+``shadow_width``, ``line_opa``, and `arc_width` in case of the styles.
+
+Therefor on this page only the differences and special considerations are highlighted.
+For a detailed description on the meaning of each field see
+
+- :ref:`style_properties`
+- Style examples
+- Canvas examples
+
+
+Base Draw Descriptor
+--------------------
+
+Each draw descriptor has a generic "base descriptor" of type ``lv_draw_dsc_base_t`` with the name ``base``. It stores useful information about which widget and part created the draw descriptor.
+See all the fields in :cpp:expr:`lv_draw_dsc_base_t`.
+
+The ``lv_obj_init_draw_...`` functions automatically initialize the fields of the base descriptor.
+
+Rectangle Draw Descriptor
+-------------------------
+
+The :cpp:expr:`lv_draw_rect_dsc_t` is a helper descriptor that combines:
+
+- fill
+- border
+- outline (essentially a border but with its own styles)
+- shadow
+- background image (an image with its own styles)
+
+into one call.
+
+``lv_obj_init_draw_rect_dsc(obj, part, &dsc);`` can be used to initialize a draw descriptor from a widget,
+and ``lv_draw_rect(layer, &dsc, area)`` draws the given rectangle in an area.
+
+Fill Draw Descriptor
+--------------------
+
+The :cpp:expr:`lv_draw_fill_dsc_t` fill descriptor is straightforward.
+It has a radius, opacity, and color to draw a rectangle.
+
+If the opacity is 0, no draw task will be created.
+
+- ``lv_draw_fill_dsc_init(&dsc)`` initializes a fill draw task.
+- ``lv_draw_fill(layer, &dsc, area)`` creates a draw task to fill an area.
+- ``lv_draw_task_get_fill_dsc(draw_task)`` gets the fill descriptor from a draw task.
+
+Gradients
+^^^^^^^^^
+
+The ``grad`` field of the fill descriptor (or :cpp_expression:`lv_grad_dsc_t` in general) can describe:
+- horizontal
+- vertical
+- skew
+- radial
+- conic
+
+gradients with multiple stops.
+
+TODO: Describe each in detail here.
```

Comment:
Nit: is it a WIP PR? If not, is it necessary to put at the documentation sections that are TODOs?

Reply:
I converted this PR to draft. I can write ~80% of this page but the vector API and the gradients probably will be documented in an other PR. 

IMO it's ok to add placeholders for these section. What do you think?